contracts/Manager.sol

```diff
@@ -209,6 +219,83 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
      * groups will be the first to have their votes withdrawn.
      */
     function deprecateGroup(address group) external onlyOwner {
+        _deprecateGroup(group);
+    }
+
+    /**
+     * @notice Checks if a group member is elected.
+     * @param groupMember The member of the group to check election status for.
+     * @return Whether or not the group member is elected.
+     */
+    function isGroupMemberElected(address groupMember) private view returns (bool) {
+        IElection election = getElection();
+
+        address[] memory electedValidatorSigners = election.electValidatorSigners();
+
+        for (uint256 i = 0; i < electedValidatorSigners.length; i++) {
+            if (electedValidatorSigners[i] == groupMember) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @notice Checks if a group meets the validator group health requirements.
+     * @param group The group to check for.
+     * @return Whether or not the group is valid.
+     */
+    function isValidGroup(address group) public view returns (bool) {
+        IValidators validators = getValidators();
+
+        // add check if group is !registered
+        if (!validators.isValidatorGroup(group)) {
+            return false;
+        }
+
+        (address[] memory members, , , , , uint256 slashMultiplier, ) = validators
+            .getValidatorGroup(group);
+
+        // check if group has no members
+        if (members.length == 0) {
+            return false;
+        }
+        // check for recent slash
+        if ((slashMultiplier) < 10**24) {
+            return false;
+        }
+        // check for majority of members are elected.
+        uint256 counter = 0;
+        for (uint256 i = 0; i < members.length; i++) {
+            if (!isGroupMemberElected(members[i])) {
+                counter += 1;
+                if (counter >= members.length / 2) {
```

Comment:
We could count members.length / 2 and store it into variable at the beginning of the cycle to save some gas

Reply:
On second thought it probably depends how many members are actually not ellected

Reply:
yeah, with only a max of 5 members, this ends up costing more gas if I store into a variable